frontend/src/components/IssueSidebar.vue

```diff
@@ -111,6 +81,40 @@
         </div>
       </template>
 
+      <h2 class="textlabel flex items-center col-span-1 col-start-1">
+        {{ $t("common.when") }}
+        <div class="tooltip-wrapper">
+          <span class="tooltip w-60">{{
+            $t("task.earliest-allowed-time-hint")
+          }}</span>
+          <!-- Heroicons name: outline/question-mark-circle -->
+          <heroicons-outline:question-mark-circle class="h-4 w-4" />
+        </div>
+      </h2>
+      <div class="col-span-2">
+        <n-date-picker
+          v-if="allowEditEarliestAllowedTime"
+          v-model:value="state.earliestAllowedTs"
+          :is-date-disabled="isDayPassed"
+          :placeholder="$t('task.earliest-allowed-time-unset')"
+          class="w-full"
+          type="datetime"
+          clearable
+          @update:value="
+            (newTimestampNs) => {
+              $emit('update-earliest-allowed-time', newTimestampNs / 1000);
+            }
+          "
+        />
+        <span v-else class="textfield col-span-2">
+          {{
+            task.earliestAllowedTs === 0
```

Comment:
If we cache a state, then we should probably use state.earliestAllowedTs. But see other comment that we may not need state at all.